vault/data_source_auth_backend.go

```diff
@@ -0,0 +1,87 @@
+package vault
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func authBackendDataSource() *schema.Resource {
+	return &schema.Resource{
+		Read: authBackendDataSourceRead,
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The auth backend mount point.",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the auth backend.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The description of the auth backend.",
+			},
+			"default_lease_ttl_seconds": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Default lease duration in seconds",
+			},
+			"max_lease_ttl_seconds": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Maximum possible lease duration in seconds",
+			},
+			"listing_visibility": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Specifies whether to show this mount in the UI-specific listing endpoint.",
+			},
+			"local": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Specifies if the auth method is local only",
+			},
+			"accessor": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The accessor of the auth backend.",
+			},
+		},
+	}
+}
+
+func authBackendDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	targetPath := d.Get("path").(string)
+
+	auths, err := client.Sys().ListAuth()
+	if err != nil {
+		return fmt.Errorf("error reading from Vault: %s", err)
+	}
+
+	for path, auth := range auths {
+		path = strings.TrimSuffix(path, "/")
+		if path == targetPath {
```

Comment:
If we dropped the "path" match here and recorded everything here, would we need to sort the auths alphabetically by name so that Terraform wouldn't think there was a constant diff due to Go's random map ordering?

Reply:
I believe Terraform is smart enough to ignore the randomness in maps and compare by key/value, TypeMap being the schema type.
Here's a test storing a `map[string]string` as a computed `schema.TypeMap`

https://github.com/hashicorp/terraform/blob/master/builtin/providers/test/resource.go